### PR TITLE
Feat: add vela debug command 

### PR DESCRIPTION
**Pull request by FogDong**
## Vela Debug
Add `vela debug` command, which can debug all the applications with `debug` policy.

All the flags:
```
Flags:
  -e, --env string         specify environment name for application
  -f, --focus string       specify the focus value to debug
  -h, --help               help for debug
  -n, --namespace string   specify the Kubernetes namespace to use
  -s, --step string        specify the step to debug
```

Usage:

```bash
vela debug <application-name>
```

First, it will ask you to choose a workflow step or component to debug:
image

Then, choose the field to debug:
image

Some example outputs:
image

Error case:
image

You can also use `--focus or -f` to show the specific value:
```bash
vela debug <application> -s <step> -f <focus>
```

Some example outputs:

image

For applications with components only, it will show the result of dry-run:
image

## Vela Up

Add `--debug` in `vela up`, if the debug flag is specified, it will add debug policy in the application automatically and jump into debug mode after the application is created.

## TODO:

- [ ] Add `--watch` flag in vela debug.
- [x] Add tests.
- [ ] Debug policy can support choosing components.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

/cc @wonderflow @Somefive @leejanee @chivalryq @yangsoon

### CODE REVIEW

**Comment by wonderflow** [comment]
this feature will go into v1.4

File: pkg/apiserver/rest/usecase/config.go [hunk 1]
```diff
@@ -368,7 +368,8 @@ func mergeTargets(currentTargets []ApplicationDeployTarget, targets []*model.Clu
 		for _, t := range targets {
 			if c.Namespace == t.Namespace {
 				hasSameNamespace = true
-				clusters := append(c.Clusters, t.ClusterName)
+				clusters := c.Clusters
+				clusters = append(clusters, t.ClusterName)
```
Comment (wonderflow) [inline pkg/apiserver/rest/usecase/config.go:372]: why changing to this?

**Review by wonderflow — COMMENTED** [review]
let's increase the test coverage to 70%

File: pkg/apiserver/rest/usecase/config.go (hunk 1 above)
Comment (FogDong) [reply pkg/apiserver/rest/usecase/config.go:372]: To fix `go lint` requires.

**Review by wonderflow — APPROVED** [review]

**Review by barnettZQG — APPROVED** [review]

**Review by Somefive — APPROVED** [review]
Generally LGTM, several recommended enhancement for future:
1. vela up command could remove the IOStream and Args parameters.
2. resourcekeeper could make the apply option args as optional ones.
3. NewWorkflow could make parameters optional as well.
We could work together to enhance them in the future.

**Review by leejanee — APPROVED** [review]

**Pull request merged**